x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -483,8 +483,10 @@ private void buildUser(SamlAttributes attributes, ActionListener<AuthenticationR
         final String name = resolveSingleValueAttribute(attributes, nameAttribute, NAME_ATTRIBUTE.name(config));
         final String mail = resolveSingleValueAttribute(attributes, mailAttribute, MAIL_ATTRIBUTE.name(config));
         UserRoleMapper.UserData userData = new UserRoleMapper.UserData(principal, dn, groups, userMeta, config);
+        logger.debug("SAML attribute mapping = [{}]", userData);
```

Comment:
This is basically the data that feeds into role mapping.
The role mapping logging will log it as well, but that debug logging may not be turned on, and it's hard to follow what's going on in this realm without it.